### DISCUSSION TITLE
Question about 3.x SQS dependency

### DISCUSSION THREAD

**Discussion by NingyuanZhang** [awspring/spring-cloud-aws#560] [Q&A] [answered]
Hi I'm using the **@SqsListener** annotation of **spring-cloud-aws-messaging** dependency to consume sqs queue and seeing the same bottleneck issue as https://github.com/awspring/spring-cloud-aws/issues/23. 

Looks like you've fixed that issue in the recent 3.0.0-M2 release, https://github.com/awspring/spring-cloud-aws/pull/374, and I'm wondering if the **SQS** dependency is ready to be used as a replacement of **spring-cloud-aws-messaging** dependency? The reason why I'm asking is I'm not able to find a **SqsMessageDeletionPolicy** class and the @SqsListene doesn't support the **deletionPolicy** parameter.

Thank you so much!
Jenny

**Comment by tomazfernandes** [comment, marked as answer]
Hi @NingyuanZhang, the `Spring Cloud AWS 3.0 SQS` integration is a complete rewrite of the feature, so probably a few adjustments will be necessary in your code.

Please refer to the [documentation](https://docs.awspring.io/spring-cloud-aws/docs/3.0.0-SNAPSHOT/reference/html/index.html#sqs-support) - the configuration you're looking for is in the [acknowledgement section](https://docs.awspring.io/spring-cloud-aws/docs/3.0.0-SNAPSHOT/reference/html/index.html#acknowledging-messages).

I hope this answers your question - feel free to ask if you have any further questions.

**Reply by NingyuanZhang** [reply]
Got it! Thanks for answering. One more quick question, I notice that the documentation URL link is for **3.0.0-SNAPSHOT**, is that a different version than 3.0.0-M2 and 3.0.0-M3? Do you have recommendations about which version I should go with?

**Reply by tomazfernandes** [reply]
Oh, sorry for the wrong link, use the [M3](https://docs.awspring.io/spring-cloud-aws/docs/3.0.0-M3/reference/html/index.html#sqs-support) please. We should have a RC soon.

**Reply by NingyuanZhang** [reply]
Cool, thanks a lot!
